Utils/HandlerUtil.py

```diff
@@ -201,24 +201,19 @@ def _parse_config(self, ctxt):
                 cms_cmd = 'openssl cms -inform DER -decrypt -recip {0} -inkey {1}'.format(cert,pkey)
                 smime_cmd = 'openssl smime -inform DER -decrypt -recip {0} -inkey {1}'.format(cert,pkey)
 
-                protected_settings_str = None
+                protected_settings_str = ''
                 for decrypt_cmd in [cms_cmd, smime_cmd]:
                     try:
-                        session = subprocess.Popen([decrypt_cmd], shell=True,
-                                                stdin=subprocess.PIPE,
-                                                stderr=subprocess.STDOUT,
-                                                stdout=subprocess.PIPE)
-                        output = session.communicate(unencodedSettings)
-                        # success only if return code is 0 and we have output
-                        if session.returncode == 0 and output[0]:
-                            protected_settings_str = output[0]
+                        output = waagent.RunSendStdin(decrypt_cmd, unencodedSettings)
+                        if output[0] == 0 and output[1]:
```

Comment:
can you please add a comment why output[0] and output[1] are special combinations and what their existance means?